src/sentinel.c

```diff
@@ -2381,6 +2376,15 @@ static int instanceLinkNegotiateTLS(redisAsyncContext *context) {
 #ifndef USE_OPENSSL
     (void) context;
 #else
+    SSL_CTX *redis_tls_ctx;
+    SSL_CTX *redis_tls_client_ctx;
+
+    if (connControl(connectionTypeTls(), CTRL_TLS_GET_CTX, (unsigned long)&redis_tls_ctx, 0, 0,0) != C_OK)
+        return C_ERR;
+
+    if (connControl(connectionTypeTls(), CTRL_TLS_GET_CLIENT_CTX, (unsigned long)&redis_tls_client_ctx, 0, 0,0) != C_OK)
+        return C_ERR;
+
```

Comment:
I think this can be somewhat misleading: it seems as if Sentinel actually supports TLS modules, but that's not the case - it only supports the OpenSSL based TLS module, because it uses `hiredis` for its client side work.

I propose these changes:
* Drop `CTRL_TLS_GET_CTX` and `CTRL_TLS_GET_CLIENT_CTX`, because they won't be useful with generic modules anyway.
* Make Sentinel support TLS only if compiled, not as a module, and then directly access these contexts.

I realize TLS+Sentinel came up before, I should have thought about this bit before...

Reply:
so Sentinel will support other connection modules, but not TLS?
it's a bit odd.

Reply:
It will support inbound modules on incoming connections only, anyway. So yes, very partial support.

Reply:
Ok, just to make things more generic, can't we use that hack of accessing these contexts directly even if TLS is loaded as a module? 

Do we need to note somewhere that (other than tls) the connection modules for sentinel only work on incoming connections? or can we skip it? (considering that there are not official connections anyway, and we only need to document tls)

Reply:
Currently, we hide all the symbols as static in tls.c, and uplayer calls TLS specific APIs by connection layer only, should we break this rule?

And I think we can rename `enum connControlType` to `enum connControlTLS`, and define `enum connControlFoo` in the future. what do you think about this?

Reply:
@pizhenwei I guess we can break this rule for Sentinel, when linking statically.

Reply:
so sentinel will support connection modules, but not tls? (so it'll be untested).
can't we break that rule for non-static linked TLS module as well?

Reply:
The TLS specific APIs is not general enough, but the global variable symbol is also not general. Currently we can call wrapper APIs to get variables instead of linking the variable symbol across source files. should we revert this part?